better-code-editing.php

```diff
@@ -158,6 +158,27 @@ public static function prep_codemirror_for_file( $file ) {
 				wp_enqueue_style( 'codemirror' );
 				wp_enqueue_style( 'codemirror-addon-lint' );
 
+				// @todo Let this be filterable?
```

Comment:
Maybe we could pass the list of rules in the `options` object that gets passed to CodeMirror, but we could have a `wp-code-editor` wrapper script, with a `wp.codeEditor.initialize( id, options )` wrapper that emulates `wp.editor.initialize()` for TinyMCE. The `wp.codeEditor.initialize` function could grab additional properties in `options`, like `csslint-rules`, and then configure CSSLint before initializing CodeMirror since the CodeMirror `options` doesn't allow for the list of rules to be defined.

Reply:
if we had a `wp.codeEditor` as well, this could provide an abstraction which would de-couple us from having to specifically support CodeMirror directly, and we could in the future switch to a different editor if we need to.

Reply:
And back to the note about being filterable, this could be part of the `$options` array which then later gets filtered `theme_editor_codemirror_opts` or `plugin_editor_codemirror_opts`, both of which could be simply a `wp_code_editor_settings` filter for parity with TinyMCE's `wp_editor_settings` filter.